### ISSUE TITLE
Rename file name from Test.go to test.go is not ok

### ISSUE THREAD

**Issue by whitefirer** [microsoft/vscode#161743]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.71.2
- OS Version: macOS Monterey 12.5.1 (21G83)

Steps to Reproduce:

1. create file named `Test.go`
2. rename the file to `test.go`
3. check the title of the file's editor tab
<img width="199" alt="image" src="https://user-images.githubusercontent.com/12084604/192210829-f6b0d8a3-fc61-4215-80e0-beb5296e39ee.png">

Is still `Test.go`, not the new name `test.go`. until reopen the file.
But `test.go` rename to `Test.go` is ok.
Also `Foo.go` reanme to `test.go` is ok.
Just `Test.go` rename to t`est.go` is not ok.

**Comment by bpasero** [comment]
Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: https://code.visualstudio.com/insiders/

**Comment by whitefirer** [comment]
> 

Ok, now it's changed when i use the insider build:
test.go rename to Test.go is not ok.
Test.go rename to test.go is ok.

It seems some intersting things here.

**Comment by bpasero** [comment]
Can you try to run `code --disable-extensions --user-data-dir <directory>` where `<directory>` is an empty folder? This will ensure Code is starting with a fresh data directory, e.g. no specific settings and without any extensions running.

**Comment by whitefirer** [comment]
> 

<img width="578" alt="image" src="https://user-images.githubusercontent.com/12084604/192284889-9fd91f04-098d-445e-92c2-74cb832b34e4.png">

As you see, nothing changed.

**Comment by bpasero** [comment]
Hm yeah I think I see it too, the first rename works fine but then renaming back does not work, is that what you see?

**Comment by bpasero** [comment]
@lramos15 this is a funny one and indeed a regression from our work to always go through the editor resolver when resolving editors and not go through the `ITextEditorService` as a shortcut.

**Steps**
* open an empty folder with 1 file `Test.go`
* open the file
* rename it from the explorer to `test.go`
* verify the tab label updates
* rename it from the explorer to `Test.go`
* 🐛 the label does not update

**Here is how it used to be**
* the rename operation will eventually hit `IEditorService.replaceEditors`
* since this is a default text editor we would directly use `ITextEditorService.createTextEditorInput`
* ⭐ we figure out that we have an existing editor input so we do not create a new one however we update its preferred resource [here](https://github.com/microsoft/vscode/blob/36b1398cfbabbe91d695717d76b818a788f7b7d9/src/vs/workbench/services/textfile/common/textEditorService.ts#L199)
* this triggers a label change event that will update the tab label

=> this works every time you rename, irrespective of going back and forward of the name

**Here is how it is today**
* we also end up in `IEditorService.replaceEditors`
* however we always now go to the editor resolver (which imho is good)
* the editor resolver however does not consider the input to be a new input, it will do an early return [here](https://github.com/microsoft/vscode/blob/36b1398cfbabbe91d695717d76b818a788f7b7d9/src/vs/workbench/services/editor/browser/editorResolverService.ts#L183-L186)
* 🐛 as such, the inputs preferred resource is never updated

Now I am actually not so sure how to fix this. I am a bit worried that updating the `FileEditorInput.matches()` method to respect resource path casing may break other behaviour that relies on the current behaviour.

Maybe the resolver should not be so clever to do an early return? Why do we do this?

**Comment by lramos15** [comment]
> Maybe the resolver should not be so clever to do an early return? Why do we do this?

Yeah maybe, I think it was to prevent creating too many inputs and calling unnecessary factories when things are not actually changing at all. I guess we could always remove the early return and see what breaks.

**Comment by bpasero** [comment]
At least for text file editors, there is caching so that we do not end up with an editor input each time, but I think the same issue already exists today: if you do not have a tab as active and the editor resolver is used to bring up that editor, we would delegate the decision whether to create an editor input or not to the provider and so its up to the provider imho do to the caching.

**Comment by lramos15** [comment]
Sorry, got caught up in other work and missed this. Will attempt a fix early in debt week for the November release as it is too risky to go in this late
